var/spack/repos/builtin/packages/cairo/package.py

```diff
@@ -25,55 +27,187 @@ class Cairo(AutotoolsPackage):
+        #  so not sure how this is supposed to work
+        variant(
+            "symbol-lookup",
+            default=False,
+            description="Not available. Enable cairo's symbol lookup functions feature",
+        )
+
+        # not currently supported variants
+        conflicts("+spectre", msg="Not currently supported")
+        conflicts("+symbol-lookup", msg="Not currently supported")
+
+        # meson seems to have assumptions about what is enabled/disabled
+        # so this protects against incompatible combinations
+        conflicts("~zlib+png", msg="+png requires +zlib")
+        conflicts("~ft+fc", msg="+fc requires +ft")
+        conflicts("+ft+fc~zlib", msg="+fc+ft requires +zlib")
+        conflicts("+fc+ft~png+zlib", msg="+ft+fc+zlib requires +png")
+
+    # meson also needs this for auto discovery of depends
     depends_on("pkgconfig", type="build")
-    depends_on("fontconfig@2.10.91:", when="+fc")  # Require newer version of fontconfig.
-    depends_on("which", type="build")
 
-    conflicts("+png", when="platform=darwin")
-    conflicts("+svg", when="platform=darwin")
-    conflicts("+shared~pic")
+    # lzo is not strictly required, but cannot be disabled and may be pulled in accidentally
+    # https://github.com/mesonbuild/meson/issues/8224
+    # https://github.com/microsoft/vcpkg/pull/38313
+    depends_on("lzo", when="@1.17.6: build_system=meson")
+
+    with when("@:1.17.6"):
+        depends_on("automake", type="build")
+        depends_on("autoconf", type="build")
+        depends_on("libtool", type="build")
+
+        depends_on("m4", type="build")
+
+        depends_on("pixman@0.36.0:", when="@1.17.2:")
+        depends_on("freetype build_system=autotools", when="+ft")
+        depends_on("fontconfig@2.10.91:", when="+fc")  # Require newer version of fontconfig.
+        depends_on("libpng", when="+png")
+        depends_on("glib")
+        depends_on("which", type="build")
+
+    with when("@1.17.8:"):
+        depends_on("meson@1.3.0:")
+        depends_on("freetype@2.13.0:", when="+ft")
+        depends_on("libpng@1.4.0:", when="+png")
+
+        depends_on("glib@2.14:", when="+gobject")
+        depends_on("pixman@0.40.0:")
+        depends_on("fontconfig@2.13.0:", when="+fc")
+        depends_on("binutils", when="+symbol-lookup")
+
+    with when("+X"):
+        depends_on("libx11")
+        depends_on("libxext")
+
+        depends_on("libxrender")
+        depends_on("libxrender@0.6:", when="@1.17.8:")
+
+        depends_on("libxcb")
+        depends_on("libxcb@1.6:", when="@1.17.8:")
+
+        depends_on("python", type="build")
 
     # patch from https://gitlab.freedesktop.org/cairo/cairo/issues/346
     patch("fontconfig.patch", when="@1.16.0:1.17.2")
     # Don't regenerate docs to avoid a dependency on gtk-doc
-    patch("disable-gtk-docs.patch", when="^autoconf@2.70:")
+    patch("disable-gtk-docs.patch", when="@:1.17.4^autoconf@2.70:")
```

Comment:
I think this might be clearer:

```suggestion
    patch("disable-gtk-docs.patch", when="build_system=meson ^autoconf@2.70:")
```

Reply:
Also while it's not possible right now, when we switch to compilers as nodes, I think this can more clearly become:

```python
    patch("disable-gtk-docs.patch", when="%autoconf@2.70:")
```

`^` (transitive link/run OR direct build dep)  will also work properly but the new semantics for `%` (direct build dep) is clearer.

Reply:
Yes I like that. n.b., that should be `build_system=autotools`. I have made the change